packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieve.ts

```diff
@@ -56,12 +57,69 @@ export class ManifestOrSourcePathGatherer
       // corresponding to the explorer path when we begin supporting multiple workspace folders
       const workspaceRootPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
       const manifestPath = path.join(workspaceRootPath, 'manifest');
-      const isManifestFile = this.explorerPath.includes(manifestPath);
-      const type = isManifestFile ? FileType.Manifest : FileType.Source;
-      return { type: 'CONTINUE', data: { filePath: this.explorerPath, type } };
+      const isManifest = this.explorerPath.includes(manifestPath);
+      const type = isManifest ? FileType.Manifest : FileType.Source;
+
+      // If the type is manifest check whether or not the user selected an actual
+      // file or just the manifest directory. If the user selected the manifest
+      // directory then check that directory for manifest files. If there's only
+      // one manifest file then use it, if there none or more than one then give
+      // them a file select dialog and make them select one.
+      if (isManifest) {
+        let manifestFile = this.explorerPath;
+        const stat = fs.lstatSync(manifestFile);
+        if (stat.isDirectory()) {
+          const manifestFileList = this.getManifestFiles(manifestFile);
+          if (manifestFileList.length === 1) {
```

Comment:
There's an edge case where a user may have one manifest file as well as one or more directories containing other manifest files in their top level manifest directory. In that case, `manifestFileList.length === 1` is true, but there could be other manifests in their manifest directory. This is a weird case, but I could see this happening if users come to really like the file picker feature and use it all the time, and then happen to have a top level manifest and also a number of other manifests organized into subfolders. They may expect to see the file picker, but instead, the top level manifest is deployed.

Reply:
@allileong @lcampos That could certainly be true but if they select the directory and do that, the acceptance criteria (pasted below, copied from the work item) states that I don't prompt in that case. That's definitely up for discussion but we should probably include @ntotten .

Acceptance criteria: 1) In the case that the manifest folder only contains 1 manifest, don't prompt, just use that manifest 2) In the event that the manifest folder contains multiple manifests, prompt to select manifest.

Reply:
This is a really good observation @allileong. I didn't think of this case.

I think you are right that we should show the picker in this case. The acceptance criteria should be amended as I missed this edge case.

Reply:
@ntotten to be clear then, if what the user selects isn't a file then always pop up the picker dialog?